sources.go

```diff
@@ -53,6 +53,25 @@ func File(name string) *Pipe {
 	return p.WithReader(f)
 }
 
+// FindFiles creates a pipe containing the files inside a directory and all its
+// subdirectories matching the supplied path, one per line.
+func FindFiles(path string) *Pipe {
+	var fileNames []string
+	err := filepath.Walk(path, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() {
+			fileNames = append(fileNames, path)
+		}
+		return nil
+	})
+	if err != nil {
```

Comment:
By the way, a common pattern is to rewrite something like:

```go
err := foo()
if err != nil {
```

as:

```go
if err := foo(); err != nil {
```

But you may feel that the function call is too long for this style to be readable. If so, I trust your judgment!

Reply:
For this to be readable, we would have to extract the function to a variable, then we can use the pattern. And I think that could be more readable.